File: supporter-service/src/main/java/fact/it/supporterservice/dto/SupporterResponse.java
package fact.it.supporterservice.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class SupporterResponse {
    private String supporterCode;
    private String firstName;
    private String lastName;
    private LocalDate birthDate;
    private String email;
    private TeamResponse teamName;
    private PlayerResponse favoritePlayer;
}

File: soccer-frontend/src/dto/SupporterResponse.ts
import { PlayerResponse } from "./PlayerResponse";
import { TeamResponse } from "./TeamResponse";
export interface SupporterResponse {
    supporterCode: string;
    firstName: string;
    lastName: string;
    birthDate: string; // Using string for LocalDate (ISO-8601 format)
    email: string;
    teamName: TeamResponse;
    favoritePlayer: PlayerResponse;
  }